Consensus.md

```diff
@@ -88,6 +98,22 @@ There are future plans to enable sidechain compatibility with Sia. This would
 allow other currencies such as Bitcoin to be spent in all the same places that
 the Siacoin can be spent.
 
+Marshalling
+-----------
+
+Many of the Sia types need to be hashed at some point, which requires having a
+consistent algorithm for marshalling types into a set of bytes that can be
+hashed. The following rules are used for hashing:
+
+ - Integers are little-endian, and are always encoded as 8 bytes.
+ - Bools are encoded as one byte, where zero is false and one is true.
+ - Variable length types such as strings are prefaced by 8 bytes containing
+   their length.
+ - Arrays and structs are encoded as their individual elements concatenated
+   together. The ordering of the struct is determined by the struct definition.
+   There is only one way to encode each type.
```

Comment:
hmm. I'm wondering now if this isn't the best approach. It might be better to manually implement `MarshalSia` and `UnmarshalSia` for all of the struct types. Not sure yet.